src/devices/DCMotor/DCMotor.cs

```diff
@@ -41,58 +41,130 @@ public virtual void Dispose()
         }
 
         /// <summary>
-        /// Constructs DCMotor class.
-        /// Depending on the settings it will pick implementation suitable for the given connection.
+        /// Creates DCMotor instance which allows to control speed in one direction.
         /// </summary>
-        public static DCMotor Create(DCMotorSettings settings)
+        /// <param name="speedControlChannel">PWM channel used to control the speed of the motor</param>
+        /// <returns>DCMotor instance</returns>
+        public static DCMotor Create(PwmChannel speedControlChannel)
         {
-            if (settings.UseEnableAsPwm)
-            {
-                if (settings.Pin0.HasValue && settings.Pin1.HasValue && settings.PwmChannel != null)
-                {
-                    return new DCMotor3Pin(
-                        settings.PwmChannel,
-                        settings.Pin0.Value,
-                        settings.Pin1.Value,
-                        settings.Controller);
-                }
-
-                throw new ArgumentException("When enable pin is used for PWM all pins and PwmChannel must be set.");
-            }
-            else
-            {
-                int? pin0 = settings.Pin0;
-                int? pin1 = settings.Pin1;
-
-                if (!pin0.HasValue && pin1.HasValue)
-                {
-                    pin0 = pin1;
-                    pin1 = null;
-                }
-
-                if (settings.PwmChannel != null)
-                {
-                    if (pin0.HasValue == pin1.HasValue)
-                    {
-                        throw new ArgumentException("Only one of Pin0 and Pin1 must be set when PWM is specified.");
-                    }
-
-                    return new DCMotor2PinNoEnable(
-                        settings.PwmChannel,
-                        pin0,
-                        settings.Controller);
-                }
-                else
-                {
-                    if (!pin0.HasValue)
-                    {
-                        // pin1 must be null now since we swap values earlier when pin0 == null
-                        throw new ArgumentException("Pin0 or Pin1 must be set when PWM is not specified.");
-                    }
-
-                    return new DCMotor2PinNoEnable(new SoftwarePwmChannel(pin0.Value, 50, 0.0, controller: settings.Controller), pin1, settings.Controller);
-                }
-            }
+            if (speedControlChannel == null)
+                throw new ArgumentNullException(nameof(speedControlChannel));
+
+            return new DCMotor2PinNoEnable(speedControlChannel, -1, null);
```

Comment:
So if you do this, seems like the DCMotor will have both a PWMChannel and a GpioController right? given that eventually the DCMotor2PinNoEnable class will call base constructor with null which will create a new GpioController. Is this controller used at all? If not then why not change the base constructor such that if null gets passed in then we don't create a new GpioController?